arangod/Replication2/ReplicatedLog/Supervision.cpp

```diff
@@ -760,20 +771,54 @@ auto executeCheckReplicatedLog(DatabaseID const& dbName,
 
   checkReplicatedLog(sctx, log, health);
 
+  bool const hasNoExecutableAction =
+      std::holds_alternative<EmptyAction>(sctx.getAction()) ||
+      std::holds_alternative<NoActionPossibleAction>(sctx.getAction());
+  // now check if there is status update
+  if (hasNoExecutableAction) {
+    // there is only a status update
+    if (sctx.isErrorReportingEnabled()) {
+      // now compare the new status with the old status
+      if (log.current && log.current->supervision) {
+        if (log.current->supervision->statusReport == sctx.getReport()) {
+          // report did not change, do not create a transaction
+          return envelope;
+        }
+      }
+    }
+  }
+
```

Comment:
I am not sure all tthis clutter is worth avoiding just calling executeAction which then does nothing.